cadctl/cmd/investigate/investigate.go

```diff
@@ -123,52 +122,27 @@ func run(_ *cobra.Command, _ []string) error {
 
 	customerAwsClient, err := managedcloud.CreateCustomerAWSClient(cluster, ocmClient)
 	if err != nil {
-		return ccam.Evaluate(cluster, err, ocmClient, pdClient, alertTypeString)
-	}
-
-	investigationResources := &investigation.Resources{AlertType: alertType, Cluster: cluster, ClusterDeployment: clusterDeployment, AwsClient: customerAwsClient, OcmClient: ocmClient, PdClient: pdClient}
-
-	run := investigation.NewInvestigation()
-
-	switch alertType {
-	case investigation.ClusterHasGoneMissing:
-		run.Triggered = chgm.InvestigateTriggered
-	case investigation.ClusterProvisioningDelay:
-		run.Triggered = cpd.InvestigateTriggered
-	default:
-		// Should never happen as GetAlertType should fail on unsupported alerts
-		return errors.New("alert is not supported by CAD")
+		return ccam.Evaluate(cluster, err, ocmClient, pdClient, alertInvestigation.Name)
 	}
 
-	eventType := pdClient.GetEventType()
-	logging.Infof("Starting investigation for %s with event type %s", alertTypeString, eventType)
+	investigationResources := &investigation.Resources{InvestigationName: alertInvestigation.Name, Cluster: cluster, ClusterDeployment: clusterDeployment, AwsClient: customerAwsClient, OcmClient: ocmClient, PdClient: pdClient}
 
-	switch eventType {
-	case pagerduty.IncidentTriggered:
-		return run.Triggered(investigationResources)
-	case pagerduty.IncidentResolved:
-		return run.Resolved(investigationResources)
-	case pagerduty.IncidentReopened:
-		return run.Reopened(investigationResources)
-	case pagerduty.IncidentEscalated:
-		return run.Escalated(investigationResources)
-	default:
-		return fmt.Errorf("event type '%s' is not supported", eventType)
-	}
+	logging.Infof("Starting investigation for %s", alertInvestigation.Name)
+	return alertInvestigation.Run(investigationResources)
 }
 
-// isAlertSupported will return the alertname as enum type of the alert if it is supported, otherwise an error
-func isAlertSupported(alertTitle string) investigation.AlertType {
+// getInvestigation will return the investigation function for the identified alert
+func getInvestigation(alertTitle string) *investigation.Investigation {
 	// We currently map to the alert by using the title, we should use the name in the alert note in the future.
 	// This currently isn't feasible yet, as CPD's alertmanager doesn't allow for the field to exist.
 
 	// We can't switch case here as it's strings.Contains.
 	if strings.Contains(alertTitle, "has gone missing") {
-		return investigation.ClusterHasGoneMissing
+		return investigation.NewInvestigation(chgm.InvestigateTriggered, "ClusterHasGoneMissing")
```

Comment:
polish: remove the "Triggered" from both functions, as we do not longer care about event type
chgm.Investigate
cpd.Investigate